lib/pipeline.js

```diff
@@ -1172,34 +1172,58 @@ class PipelineModel extends BaseModel {
 
         // Calculate event duration by using max endTime - min startTime of builds
         // that belong this event
-        const eventDuration = async (event) => {
-            const builds = await event.getBuilds();
+        const eventMetrics = async (event) => {
+            const buildMetrics = await event.getBuildMetrics();
 
             // Somehow this event doesn't have any builds
-            if (!builds || builds.length === 0) {
-                return 0;
+            if (!buildMetrics || buildMetrics.length === 0) {
+                return {
+                    duration: 0,
+                    queuedTime: 0,
+                    imagePullTime: 0
+                };
             }
 
-            let minStartTime = new Date(builds[0].startTime);
-            let maxEndTime = new Date(builds[0].endTime);
-
-            for (let i = 1; i < builds.length; i += 1) {
-                const s = new Date(builds[i].startTime);
-                const e = new Date(builds[i].endTime);
+            let minStartTime = new Date(buildMetrics[0].startTime);
+            let maxEndTime = new Date(buildMetrics[0].endTime);
+            let eventStatus = buildMetrics[0].status;
```

Comment:
Is it possible there's no status if `[skip ci]` was used?

Reply:
`[skip ci]` won't have a build so it will not be part of `buildMetrics`